### DISCUSSION TITLE
Show models based on a users role

### DISCUSSION THREAD

**Discussion by marvimarv** [danny-avila/LibreChat#2176] [Help Wanted]
Hello everyone. I am working on a small admin-backend for my university which already allows to set tokenCredits so we can manage how much tokens a user can use. I am now looking for the place inside the code where the fetching of the models is done to intercept this and only show models based on a users role that can be set from the backend as well. 
At the moment my backend is connected to the MongoDB database directly and just setting the specified tokens there. I would do the same for the roles and basically just need to place a function in the code where the models are being set to be dependent on a role.

It would be great for someone to point me into the right direction. Of course I will be sharing my finished backend if anyone would like to use it :-)

Thanks in advance!

**Comment by danny-avila** [comment]
Role based access controls are planned and the next big project I'm working on for the repo.

>  I am working on a small admin-backend

So am I :)

See https://librechat.ai

**Reply by marvimarv** [reply]
I saw that you are working on it and that’s great! Unfortunately our university needs a solution until your backend is ready, so I am building something small during that as there is no timeframe on when the admin panel is released. Which is totally fine as I know it is a lot of work! I am just trying to find a solution during this period :-)

Would you be able to help me with my question ?

Thanks in advance!

**Comment by AxelCont** [comment]
Hello.
I wanted to know if there is any news on this issue.
Is there anything that could be considered?
Thanks

**Comment by laelhalawani** [comment]
It seems it was implemented here https://github.com/danny-avila/LibreChat/pull/7804
